catalyst-gateway/bin/src/db/index/queries/caches/rbac/chains.rs

```diff
@@ -0,0 +1,68 @@
+//! RBAC Persistent Chains Cache.
+
+use catalyst_types::catalyst_id::CatalystId;
+use get_size2::GetSize;
+use moka::policy::EvictionPolicy;
+use rbac_registration::registration::cardano::RegistrationChain;
+
+use crate::{service::utilities::cache::Cache, settings::Settings};
+
+/// RBAC Chains Cache.
+pub(crate) struct ChainsCache {
+    /// Interior cache type.
+    inner: Cache<CatalystId, RegistrationChain>,
+}
+
+impl ChainsCache {
+    /// Name for cache
+    const CACHE_NAME: &str = "RBAC Persistent Chains Cache";
+
+    /// Function to determine cache entry weighted size.
+    fn weigher_fn(k: &CatalystId, v: &RegistrationChain) -> u32 {
+        let k_size = GetSize::get_size(&k);
+        let v_size = GetSize::get_size(&v);
+        k_size.saturating_add(v_size).try_into().unwrap_or(u32::MAX)
+    }
+
+    /// New Chains Cache instance.
+    ///
+    /// # Arguments
+    /// * `is_persistent` - If the `CassandraSession` is persistent.
+    ///
+    /// Disables the cache for Volatile sessions (`is_persistent` is `false`).
+    pub(crate) fn new(is_persistent: bool) -> Self {
+        let max_capacity = if is_persistent {
+            Settings::rbac_cfg().persistent_chains_cache_size
+        } else {
+            0
+        };
+        Self {
+            inner: Cache::new(
+                Self::CACHE_NAME,
+                EvictionPolicy::lru(),
+                max_capacity,
+                Self::weigher_fn,
+            ),
+        }
+    }
+
+    /// Get an entry from the cache
+    pub(crate) fn get(&self, key: &CatalystId) -> Option<RegistrationChain> {
```

Comment:
I think these methods could be exposes just by implementing `Deref` trait.
```rust
impl Deref for ChainsCache {
 type Target = Cache<CatalystId, RegistrationChain>;
 ...
}
```

Reply:
I would prefer to keep the API explicit for each cache, and not expose the entire cache functionality. For example, the `ChainsCache` does not need to be cleared, etc.